### PR TITLE
Added tests for AUTH module

### PR DESCRIPTION
**Pull request by MaraDascalu**
Added tests for :
-> user with invalid email
-> user with invalid password
-> user with both email and password vallid

### CODE REVIEW

File: lib/util/AuthApp/actions.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { LoginHeadingTitle } from "./constants.js";
+import { LoginHeading, SignInHeading, PasswordHeading, EmailErrorMessage, PasswordErrorMessage} from "./constants.js";
+import { USER } from "../../../config/user.conf.js"
```
Comment (MaraDascalu) [inline lib/util/AuthApp/actions.js:2]: Maybe we can move this import in authenticate.js file?

File: lib/util/AuthApp/actions.js [hunk 2]
```diff
@@ -17,10 +18,61 @@ export default class AuthAppActions extends AuthAppGetters {
...
     }
+
+    async loginWithExistingAcc() {
+        await (await this.getTitleHeading()).waitForDisplayed({ timeout: 3000});
+        await expect(await this.getTitleHeading()).toHaveText(SignInHeading);
+        await (await this.getExistingAccountButton()).waitForClickable({ timeout: 30000 });
+        await (await this.getExistingAccountButton()).click();
+    }
+
+    // finished with "s.unibuc.roo"
```
Comment (MaraDascalu) [inline lib/util/AuthApp/actions.js]: Please don't forget to tests the other scenarios that we've talked about

File: lib/util/AuthApp/getters.js [hunk 3]
```diff
@@ -10,7 +10,41 @@ export default class AuthAppGetters extends BasePage {
...
     }
-}
\ No newline at end of file
+
+    async getTitleHeading() {
+        // toDo create a constant for "Create an account"
+        return (await $('//div[@role=\'heading\']'));
+    }
+
+    // async getExistingAccountButton() {
```
Comment (MaraDascalu) [inline lib/util/AuthApp/getters.js]: If we don't find a scenario to replicate the "Login with existing account" flow, this should be removed

File: test/AUTH/General/authenticate.js [hunk 4]
```diff
@@ -3,14 +3,35 @@ import AuthAppActions from "../../../lib/util/AuthApp/actions.js";
...
         await AuthApp.successfulOpen();
         await AuthApp.navigateToLogin();
    }); 
 
-   it('for user with role STUDENT', async () => {
-        console.log("It is working!!!");
-   });
-});
\ No newline at end of file
+//    it('for existing account', async () => {
```
Comment (MaraDascalu) [inline test/AUTH/General/authenticate.js]: Let's not forget to try to reproduce this scenario

**Comment by MaraDascalu** [comment]
Also, maybe we can change the structure of the USER constant to something like this:
`USER = {
   VALID: {
      EMAIL: ...
      PASSWORD: ....
      NAME: {
         FIRST: ...
         LAST:...
      }
   INVALID: {
      EMAIL:...
      PASSWORD: ...
   }
}`

File: lib/util/AuthApp/actions.js [hunk 5]
```diff
@@ -17,10 +18,62 @@ export default class AuthAppActions extends AuthAppGetters {
...
+    async loginWithInvalidEmail() {
+        await browser.pause(3000);
```
Comment (MaraDascalu) [inline lib/util/AuthApp/actions.js]: Please remove this line

File: lib/util/AuthApp/actions.js [hunk 6]
```diff
@@ -67,13 +66,50 @@ export default class AuthAppActions extends AuthAppGetters {
...
     }
 
+    /*
+    ** Reset account
+    */
 
+    async cannotAccessAccount() {
+        await (await this.getTitleHeading()).waitForDisplayed({ timeout: 3000 });
+        await expect(await this.getTitleHeading()).toHaveText(SignInHeading);
+
```
Comment (MaraDascalu) [inline lib/util/AuthApp/actions.js]: Maybe this empty lines should be removed. Or if you want to adopt this style, all functions should respect this

File: lib/util/AuthApp/constants.js
Comment (MaraDascalu) [inline lib/util/AuthApp/constants.js:1]: I think it is a better idea to group constants according to their usage. For example: the ones that are used for Reset account page can be grouped like this:
const RESET_ACCOUNT = {
    INFO: '...',
    HEADING: '...'
}

File: test/AUTH/General/authenticate.js
Comment (MaraDascalu) [inline test/AUTH/General/authenticate.js:1]: Similar to the beforeEach block were we added some extra steps before the tests, we should add an afterEach block were we logout the user (if login is successful)

File: test/AUTH/General/resetAccount.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+describe('User should be able to reset', async() => {
+   beforeEach('Navigate to login page for Moodle', async () => {
+      await AuthApp.open();
+      await AuthApp.successfulOpen();
+      await AuthApp.navigateToLogin();
+   }); 
+
+   it('work email', async() => {
+      await AuthApp.resetWorkAccount();
+      await AuthApp.successfulOpen("passwordreset.microsoftonline.com");
```
Comment (MaraDascalu) [inline test/AUTH/General/resetAccount.js:15]: The string that is passed to successfulOpen() should be included in a constant file

**Review by alexandra-udristoiu — APPROVED** [review]

**Pull request merged**